config/config.go

```diff
@@ -474,6 +483,34 @@ func (c *ServersetSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
 	return checkOverflow(c.XXX, "serverset_sd_config")
 }
 
+// MarathonSDConfig is the configuration for services running on Marathon
+type MarathonSDConfig struct {
+	Name           string   `yaml:"name,omitempty"`
```

Comment:
We strongly prefer having a single way of doing things over convenience to avoid accumulating complexity. 

The source names are prefixed by the scrape configs' default job names, which are unique. So no issue across multiple scrape configs.
Now if we were to configure multiple clusters within the same scrape config it could indeed be an issue. But this is true for Serversets (@brian-brazil), DNS and everything else.

This can be fixed by throwing the number of the SD configuration into the source. This can be done transparently so the user does not have to worry about it.

Summed up, I'd prefer to remove the `Name` and `UseJobPrefix` options unless you see a reason not addressed by the above.